tests/functional/events/09-task-event-mail.t

```diff
@@ -49,15 +49,16 @@ run_ok "${TEST_NAME_BASE}-validate" \
 workflow_run_ok "${TEST_NAME_BASE}-run" \
     cylc play --reference-test --debug --no-detach ${OPT_SET} "${WORKFLOW_NAME}"
 
-contains_ok "${TEST_SMTPD_LOG}" <<__LOG__
-b'retry: 1/t1/01'
-b'succeeded: 1/t1/02'
-b'see: http://localhost/stuff/${USER}/${WORKFLOW_NAME}/'
-__LOG__
+grep_ok 'retry: 1/t1/01' "${TEST_SMTPD_LOG}"
+grep_ok 'succeeded: 1/t1/02' "${TEST_SMTPD_LOG}"
+grep_ok "see: http://localhost/stuff/${USER}/${WORKFLOW_NAME}/" "${TEST_SMTPD_LOG}"
```

Comment:
The previous test ensured that all three were contained in the same entry (i.e. email) which may have been a useful test.

I had to convert a few of these in https://github.com/cylc/cylc-flow/pull/5769 and used the `-P` option to grep. I didn't see performance issues when I tried it.

Reply:
Yes - this change seems unecessary anyway. No idea what's going on.

Reply:
I don't think that it did: On master, running this manually I get two emails, one with the first line, one with the second and both with the third. This is also what I see in the test.

Moreover I don't think contains_ok implies that the three lines are a contiguous bloc - which raises the question - why is this failing. If that's true, however, this change ought to be unnecessary.

On closer examination it looks like its the `b''` which is the problem - which makes sense given the change in smtpd mocking engine.

Reply:
> I don't think contains_ok implies a contiguous block

Correct.